### DISCUSSION TITLE
Server Port

### DISCUSSION THREAD

**Discussion by DevBlokJochem** [Minestom/Minestom#821] [Help]
Hello, is there a way to get the port that is defined in the server.properties?

**Comment by Flamgop** [comment]
You'd have to make your own `server.properties` using java's Properties system, then get from it that way.

**Reply by DevBlokJochem** [reply]
I think that my hosting is creating that file for me. thats the reason why Im confused.

**Reply by Flamgop** [reply]
well in that case, you need to create a FileInputStream and use Properties#load, then use Properties#getProperty to get the value, then use Integer#parseInt to get your port from output of Properties#getProperty and then feed that to MinecraftServer#start

**Comment by MrlnHi** [comment]
Minestom does not support server.properties files natively, you'd have to parse it yourself

**Reply by DevBlokJochem** [reply]
I think that my hosting is creating that file for me. thats the reason why Im confused.

**Comment by DevBlokJochem** [comment]
is there a way to get the allocated port from pterodactyl for the server?

**Reply by ZakShearman** [reply]
not directly but in pterodactyl egg configurations you can use the port as a variable in some of the places there. You might also be able to access that variable in start commands.
